src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {
+  useState,
+  useEffect,
+  useCallback,
+} from 'react';
+import classNames from 'classnames';
+
+import { Header } from '../Header';
+import { Footer } from '../Footer';
+import { UserWarning } from '../UserWarning';
+import { TodoList } from '../TodoList';
+import { Error } from '../Error';
+
+import { Todo } from '../../types/Todo';
+import { getTodos, toogleTodo, deleteTodo } from '../../api/todos';
+import { useLocalStorage } from '../../utils/useLocalStorage';
+import { User } from '../../types/User';
+import { warningTimer } from '../../utils/warningTimer';
+
+export const TodoApp: React.FC = () => {
+  /* eslint-disable @typescript-eslint/no-non-null-assertion */
+  const [todos, setTodos] = useState<Todo[]>(JSON.parse(
```

Comment:
Same here, can we rewrite it using `useLocalStorage`, it should more convenient 

Plus avoid `!` in similar way as in previous comment